src/cpprealm/schedulers/default_schedulers.cpp

```diff
@@ -0,0 +1,138 @@
+#include <cpprealm/schedulers/default_schedulers.hpp>
+
+#include <realm/object-store/util/scheduler.hpp>
+
+#if defined(REALM_HAVE_UV) && REALM_HAVE_UV
+#include <uv.h>
+#endif
+
+namespace realm {
+
+    void realm_core_scheduler::invoke(std::function<void()> &&fn) {
+        s->invoke(std::move(fn));
+    }
+
+    [[nodiscard]] bool realm_core_scheduler::is_on_thread() const noexcept {
+        return s->is_on_thread();
+    }
+
+    bool realm_core_scheduler::is_same_as(const scheduler *other) const noexcept {
+        if (auto o = dynamic_cast<const realm_core_scheduler *>(other)) {
+            return o->s->is_same_as(this->s.get());;
+        }
+        return false;
+    }
+
+    [[nodiscard]] bool realm_core_scheduler::can_invoke() const noexcept {
+        return s->can_invoke();
+    }
+
+    realm_core_scheduler::operator std::shared_ptr<util::Scheduler>() {
+        return s;
+    }
+
+#if defined(REALM_HAVE_UV) && REALM_HAVE_UV
+    class invocation_queue {
+    public:
+        void push(std::function<void()>&&);
+        void invoke_all();
+
+    private:
+        std::mutex m_mutex;
+        std::vector<std::function<void()>> m_functions;
+    };
+
+    void invocation_queue::push(std::function<void()>&& fn)
+    {
+        std::lock_guard lock(m_mutex);
+        m_functions.push_back(std::move(fn));
+    }
+
+    void invocation_queue::invoke_all()
+    {
+        std::vector<std::function<void()>> functions;
+        {
+            std::lock_guard lock(m_mutex);
+            functions.swap(m_functions);
+        }
+        for (auto&& fn : functions) {
+            fn();
+        }
+    }
+
+    class uv_scheduler final : public realm::scheduler {
```

Comment:
Instead of having a whole other implementation here, couldn't we just implement make_uv() in terms of `realm_core_scheduler(std::make_shared<::realm::util::UvScheduler>())`, i.e. borrow Core's implementation? Core's implementation always assumes the default uv loop, but it can be refactored to accept a uv_loop argument.